commit 87253d1b4f2b5a29bdfc6275b9fb52a47d72df64
Author: Ian Campbell <ijc@hellion.org.uk>
Date:   Tue Feb 19 11:12:30 2008 +0000

    x86: boot protocol updates
    
    Also update field names to simply payload_{offset,length} so as to not rule
    out uncompressed images.
    
    Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
    Cc: H. Peter Anvin <hpa@zytor.com>
    Cc: Jeremy Fitzhardinge <jeremy@goop.org>
    Cc: virtualization@lists.linux-foundation.org
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff --git a/Documentation/i386/boot.txt b/Documentation/i386/boot.txt
index 05c24df..2eb1610 100644
--- a/Documentation/i386/boot.txt
+++ b/Documentation/i386/boot.txt
@@ -170,8 +170,8 @@ Offset	Proto	Name		Meaning
 0238/4	2.06+	cmdline_size	Maximum size of the kernel command line
 023C/4	2.07+	hardware_subarch Hardware subarchitecture
 0240/8	2.07+	hardware_subarch_data Subarchitecture-specific data
-0248/4	2.08+	compressed_payload_offset
-024C/4	2.08+	compressed_payload_length
+0248/4	2.08+	payload_offset	Offset of kernel payload
+024C/4	2.08+	payload_length	Length of kernel payload
 
 (1) For backwards compatibility, if the setup_sects field contains 0, the
     real value is 4.
@@ -514,22 +514,24 @@ Protocol:	2.07+
 
   A pointer to data that is specific to hardware subarch
 
-Field name:	compressed_payload_offset
+Field name:	payload_offset
 Type:		read
 Offset/size:	0x248/4
 Protocol:	2.08+
 
   If non-zero then this field contains the offset from the end of the
-  real-mode code to the compressed payload. The compression format
-  should be determined using the standard magic number, currently only
-  gzip is used.
+  real-mode code to the payload.
+
+  The payload may be compressed. The format of both the compressed and
+  uncompressed data should be determined using the standard magic
+  numbers. Currently only gzip compressed ELF is used.
   
-Field name:	compressed_payload_length
+Field name:	payload_length
 Type:		read
 Offset/size:	0x24c/4
 Protocol:	2.08+
 
-  The length of the compressed payload.
+  The length of the payload.
 
 **** THE IMAGE CHECKSUM
 
diff --git a/arch/x86/boot/header.S b/arch/x86/boot/header.S
index 40c91bb..6d2df8d 100644
--- a/arch/x86/boot/header.S
+++ b/arch/x86/boot/header.S
@@ -224,10 +224,8 @@ hardware_subarch:	.long 0			# subarchitecture, added with 2.07
 
 hardware_subarch_data:	.quad 0
 
-compressed_payload_offset:
-			.long input_data
-compressed_payload_length:
-			.long input_data_end-input_data
+payload_offset:		.long input_data
+payload_length:		.long input_data_end-input_data
 
 # End of setup header #####################################################
 
